bpf/bpf_xdp.c

```diff
@@ -25,6 +25,11 @@
  */
 #define SKIP_SRV6_HANDLING
 
+/* Controls the inclusion of egress-related nodeport sections in the object file.
+ * As XDP doesn't have an egress hook, they are not needed.
+ */
+#define SKIP_NODEPORT_EGRESS_HANDLING		1
```

Comment:
I know this is just adding an extra condition to an existing declare_tailcall_if, but.. wouldn't it make more sense to split this into `nodeport_ingress.h` and `nodeport_egress.h` and take `nodeport_egress.h` out of the include chain of bpf_xdp?

Essentially, I want a way out of declare_tailcall_if because it feels like we're slowly painting ourselves into a corner. We could technically start doing this tail call pruning in user space today, but we need a clear way of defining what needs to be pruned when.

Reply:
ack, this was just the approach with the least impact on backports. Happy to revisit once we get serious about ripping out the conditional tailcalls, I like your suggestion!

Reply:
-> https://github.com/cilium/cilium/pull/35474